apps/backend/src/applications/utils.ts

```diff
@@ -0,0 +1,4 @@
+import { Cycle } from './dto/cycle.dto';
+import { Semester } from './types';
+
+export const getCurrentCycle = () => new Cycle(2023, Semester.FALL);
```

Comment:
```suggestion
export const getCurrentCycle = () => new Cycle(2023, Semester.FALL);

export const getLatestApplication = (applications: Application[]) => { 
    if (applications.length === 0) {
        throw new BadRequestException('Application not found');
    }
    let latestApplication = applications[0];
    
    for (let j = 1; j < applications.length; j++) {
        const application = applications[j];
        const cycle = plainToClass(Cycle, application.cycle);
        
        if (cycle.year > latestApplication.cycle.year) {
            latestApplication = application;
        }
        if (cycle.year === latestApplication.cycle.year) {
            if (cycle.semester === Semester.FALL) {
                latestApplication = application;
            } else {
                continue; // if the current application is the same year as the latest application and the current application is not the fall semester, it must be the spring semester and therefore not the latest application
            }
        }
        // if the current application has a year less than the latest application, it is not the latest
    }
    return latestApplication;
}
```
I did not test this at all, but having something like this would be nice!